packages/core/lib/testing/soliditytest.js

```diff
@@ -1,178 +1,172 @@
-      series(
-        [
-          runner.initialize.bind(runner),
-          self.compileNewAbstractInterface.bind(this, runner),
-          self.deployTestDependencies.bind(
-            this,
-            abstraction,
-            dependency_paths,
-            runner
-          )
-        ],
-        done
-      );
+    suite.beforeAll("prepare suite", async function() {
+      await runner.initialize.bind(runner)();
+      await self.compileNewAbstractInterface.bind(this)(runner);
+      await self.deployTestDependencies.bind(
+        this
+      )(abstraction, dependencyPaths, runner);
     });
 
-    suite.beforeEach("before test", function(done) {
-      runner.startTest(this, done);
+    suite.beforeEach("before test", async function() {
+      await runner.startTest(this);
     });
 
-    // Function that decodes raw logs from unlinked third party assertion
-    // libraries and returns usable TestEvent logs
-    function decodeTestEvents(result) {
-      if (result.logs.length) return result.logs;
-
-      const logs = [];
-      const signature = web3Utils.sha3("TestEvent(bool,string)");
-
-      result.receipt.logs.forEach(log => {
-        if (log.topics.length === 2 && log.topics[0] === signature) {
-          const decoded = {
-            event: "TestEvent",
-            args: {
-              result: abi.decodeLog(["bool"], log.topics[1], log.topics)[0],
-              message: abi.decodeLog(["string"], log.data, log.topics)[0]
+    // Function that checks transaction logs to see if a test failed.
+    async function checkResultForFailure(result) {
+      const logs = result.receipt.rawLogs;
+      for (const log of logs) {
+        const decodings = await runner.decoder.decodeLog(log);
+        for (const decoding of decodings) {
+          //check: is this a TestEvent?
+          //note: we don't check the argument names
+          if (
+            decoding.abi.name === "TestEvent" &&
+            decoding.arguments.length === 2 &&
+            decoding.arguments[0].value.type.typeClass === "bool" &&
+            decoding.arguments[0].indexed &&
+            decoding.arguments[1].value.type.typeClass === "string" &&
+            !decoding.arguments[1].indexed
+          ) {
+            //if so: did the test fail?
+            if (!decoding.arguments[0].value.value.asBoolean) {
+              //if so: extract the message
+              let messageDecoding = decoding.arguments[1].value;
+              let message;
+              switch (messageDecoding.value.kind) {
+                case "valid":
+                  message = messageDecoding.value.asString;
+                  break;
+                case "malformed":
+                  //use buffer to convert hex to string
+                  //(this causes malformed UTF-8 to become U+FFFD)
+                  message = Buffer.from(
+                    messageDecoding.value.asHex.slice(2),
+                    "hex"
+                  ).toString();
```

Comment:
Does it make sense at all to provide a fallback "default" case here?

Reply:
No, those are the only two cases, as specified by the format.